### DISCUSSION TITLE
Dynamic column span based on a dropdown update

### DISCUSSION THREAD

**Discussion by george-luciditylab** [filamentphp/filament#9248] [Help]
I'm wondering if someone can point me in the right direction with this, I have been getting a little lost in the documentation trying to find a solution. The issue is as follows:

I have a builder component with 12 columns. Each builder block can span up to 12 of these columns. Each block has a 'layout' Select component with each option matching a desired column span value. What I am trying to achive is that when the Select component is changed I would like to update the column span of the block.

I am assuming the issue is caused by the path to the 'layout' component and I'm not too sure how to resolve. Any pointers will be greatly appreciated :)

**Here is stripped down example of my code:**

`BuilderAlias::make('page_content')`
`    ->columns(12)`
`    ->blocks([`
`        BuilderAlias\Block::make('text_block')`
`            ->label('Text Block')`
`            ->columnSpan(function (Get $get) {`
`                return (int)$get('layout');`
`            })`
`            ->schema([`
`                Section::make('Block Settings')`
`                    ->schema([`
`                        Select::make('layout')`
`                            ->label('Block Width')`
`                            ->options([`
`                                '12' => '100% Width',`
`                                '10' => '83% Width',`
`                                '8' => '75% Width',`
`                                '7' => '66% Width',`
`                                '6' => '50% Width',`
`                                '4' => '33% Width',`
`                                '3' => '25% Width'`
`                            ])`
`                            ->default('12')`
`                            ->live()`
`                    ])`
`            ])`
`    )]`

**Comment by VincvanAs** [comment]
Hi maybe accessing the state works:

`->columnSpan(function (?array $state): int {
        if ($state === null) {
            return 12;
        }
return $state['layout'];

})`

Didn't tried it tho

**Reply by george-luciditylab** [reply]
Hey, I tried your solution but have had no joy. I think the issue may be the builder not supporting columns :)

**Reply by malithmk** [reply]
Hi Did you able to solve this issue?
